src/pom.ts

```diff
@@ -1,17 +1,119 @@
-import { RequestHandler } from "express";
+import {RequestHandler} from "express";
+import {
+    CurseForgeError,
+    fetchAllModFiles,
+    fetchModFile,
+    fetchModMetadata,
+    isLatinLetter,
+    partition,
+    wasErroneous
+} from "./util";
+import {ModFileDependency, ModFileDependencyType, ModFileMetadata, ModMetadata} from "./modmetadata";
 
-const pom: RequestHandler = async (req, res) => {
-  const { descriptor, fileIds } = req.params
-  return res.send(
-    `<?xml version="1.0" encoding="UTF-8"?>
+export const generatePom = async (id: any, fileId: string, descriptor: string): Promise<string | CurseForgeError> => {
+    const metadata = await fetchModMetadata(id)
+    if (wasErroneous(metadata)) {
+        return metadata
+    }
+
+    const file = await fetchModFile(id, fileId)
+    if (wasErroneous(file)) {
+        return file
+    }
+    const publicationMillis = filePublicationMillis(file);
+    const dependencies = file.dependencies
+        .filter((it: ModFileDependency) =>
+            it.relationType === ModFileDependencyType.RequiredDependency ||
+            it.relationType === ModFileDependencyType.Include ||
+            it.relationType === ModFileDependencyType.Tool)
+        .map(async (dep: ModFileDependency) => await resolveDependencyFile(dep.modId + "", publicationMillis, file.gameId, file.gameVersions))
+
+    const dependencyStringsOrError = await Promise.all(dependencies);
+    for (const e of dependencyStringsOrError) {
+        if (wasErroneous(e)) {
+            return e
+        }
+    }
+    const dependenciesString = dependencyStringsOrError
+        .map((it) => it as ResolvedModDependency)
```

Comment:
This map is a bit ugly. This would be solved with the `CurseForgeResponse<T>` type described below

Reply:
The issue is returning early if one of the items isn't successful. Even with the CurseForgeResponse<T> (which is so much cleaner than what I had lol), you have to make this same cast... Am going to leave it as is for now.